### ISSUE TITLE
ng test not reporting TypeScript errors

### ISSUE THREAD

**Issue by meiram-tr** [angular/angular-cli#24793]
### Command

test

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

same issue as described in 
[ng test not reporting TypeScript errors #5863](https://github.com/angular/angular-cli/issues/5863)

reproduced on 14.2.1

### Minimal Reproduction

* Perform a fresh install with ng new
* Add the following test to src/app/app.component.spec.ts

> it('should not compile if there are TypeScript errors', () => {
>   const myFunc = (arg: string) => {};
>   myFunc(1);
> });

* run ng test 

see errors of typescript but result is success 0/0

### Your Environment

```text
ngular CLI: 14.2.1
Node: 16.13.1
Package Manager: npm 8.1.2 
OS: darwin x64

Angular: 14.2.0
... animations, cdk, common, compiler, compiler-cli, core
... elements, forms, language-service, material
... platform-browser, platform-browser-dynamic, router, upgrade

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.1
@angular-devkit/build-angular   14.2.1
@angular-devkit/core            14.2.1
@angular-devkit/schematics      14.2.1
@angular/cli                    14.2.1
@schematics/angular             14.2.1
ng-packagr                      14.0.3
rxjs                            7.5.6
typescript                      4.6.4
```

**Comment by alan-agius4** [comment]
Hi @meiram-tr, I am a bit confused by the report.

You mention that TypeScript errors are not reported but in the reproduction steps you mentioned that TypeScript errors are reported. 

```
$ ng t
✔ Browser application bundle generation complete.

Error: src/app/app.component.spec.ts:15:12 - error TS2345: Argument of type 'number' is not assignable to parameter of type 'string'.

15     myFunc(1);
              ~
28 02 2023 14:55:37.080:WARN [karma]: No captured browser, open http://localhost:9876/
28 02 2023 14:55:37.093:INFO [karma-server]: Karma v6.4.1 server started at http://localhost:9876/
28 02 2023 14:55:37.094:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
28 02 2023 14:55:37.098:INFO [launcher]: Starting browser Chrome
28 02 2023 14:55:43.778:INFO [Chrome 110.0.0.0 (Linux x86_64)]: Connected on socket lVE42Z1MJ1Nho6YIAAAB with id 76198351
28 02 2023 14:55:43.964:WARN [web-server]: 404: /_karma_webpack_/main.js
Chrome 110.0.0.0 (Linux x86_64): Executed 0 of 0 SUCCESS (0.01 secs / 0 secs)
TOTAL: 0 SUCCESS
```

**NB:** `TOTAL: 0 SUCCESS` is expected when running in watch mode. This is because test are not executed until the type error is fixed.


What are you expected?

**Comment by meiram-tr** [comment]
HI @alan-agius4 thanks for quick reply the the problem is that ng test process finish with success, and it not running the specs at all. expected result that the process will exist with error as the unit tests failed to run due to typescript error. instead we get success.... hope it is more clearer

**Comment by alan-agius4** [comment]
The above behaviour is expected when running in watch mode which is the default. In this case `ng test` process does not finish with success or failure, but rather is watching the files and waiting for the TypeScript errors to be fixed.

To exit with a terminate the process on error and exit with non zero error code you need to run `ng test` in non watch mode.

```
ng test --no-watch
```

**Comment by meiram-tr** [comment]
@alan-agius4 we are setting --no-watch but still the process continue without non zero error it just prints the error.
The command line:
 ng test [library-name] --no-watch --no-progress --browsers ChromeHeadless

![image](https://user-images.githubusercontent.com/7975804/222188644-d5eea8fb-78ef-48b6-8731-e5a3a36783fc.png)

**Comment by alan-agius4** [comment]
Apologies, that issue was only fixed in version. 

See: https://github.com/angular/angular-cli/issues/23751

**Comment by meiram-tr** [comment]
Thank @alan-agius4 for your very fast  response, do you think it will be merged also to 14.2.X ?

**Comment by alan-agius4** [comment]
Version 14 is currently in LTS and it only received security related updates.

See: https://angular.io/guide/releases#lts-fixes
